### PR TITLE
[clang][NFC] Fix the static assertion in 4797437

### PR DESCRIPTION
**Pull request by ziqingluo-90**
In the previous commit 4797437463e63ee289a1ff1904cfb7b2fe6cb4c2, I used `llvm::isInt<NumStmtBits>(StmtClass::LAST##Class)` to test if `StmtClass` is strictly bounded by the an unsigned integer of 'NumStmtBits'.  That is incorrect as `llvm::isInt` tests for signed integers. This commit fixes it.

### CODE REVIEW

File: clang/include/clang/AST/Stmt.h
Comment (jroelofs) [inline clang/include/clang/AST/Stmt.h:118]: sorry about that... how about `isUInt` instead?

**Review by jroelofs — APPROVED** [review]

File: clang/include/clang/AST/Stmt.h
Comment (jroelofs) [inline clang/include/clang/AST/Stmt.h:117]:
```suggestion
  static_assert(llvm::isUInt<NumStmtBits>(                                     \
```

**Comment by jroelofs** [comment]
may as well address @nikic's post-commit suggestion to move this out of the header, and into a source file, while you're here: https://github.com/llvm/llvm-project/pull/120341#discussion_r1893822845

**Comment by ziqingluo-90** [comment]
> [#120341 (comment)](https://github.com/llvm/llvm-project/pull/120341#discussion_r1893822845)

oops, I saw this after I hit the merge button.  I will do it in a follow-up commit.

**Pull request merged**
